applications/PoromechanicsApplication/CMakeLists.txt

```diff
@@ -131,12 +132,14 @@ else()
         ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_normal_fluid_flux_3D_diff_order_condition.cpp
     )
 
-    ## define library Kratos which defines the basic python interface
-    add_library(KratosPoromechanicsApplication SHARED ${KRATOS_POROMECHANICS_APPLICATION_SOURCES})
-    target_link_libraries(KratosPoromechanicsApplication KratosCore KratosSolidMechanicsApplication)
+    #target_link_libraries(KratosPoromechanicsApplication KratosCore KratosSolidMechanicsApplication)
 
 endif()
 
+## define library Kratos which defines the basic python interface
+pybind11_add_module(KratosPoromechanicsApplication MODULE ${KRATOS_POROMECHANICS_APPLICATION_SOURCES})
+target_link_libraries(KratosPoromechanicsApplication PRIVATE KratosCore)
```

Comment:
@RiccardoRossi isn't it necessary that we add here the link to other applications in Kratos (like I had before with the SolidMechanicsApplication)?

Reply:
In principle it is a bad thing that you have a compile-time dependency on the other apps. If you cannot avoid, the solution we found was to define a "core" lib inside the app and to link to that instead of to the python module.

take a look to the latest commits into the pybind branch since they go in this direction

Reply:
Ok, thanks. I'll look into it this afternoon and commit the necessary changes in this PR